pandas/tests/frame/test_analytics.py

```diff
@@ -1965,3 +1899,96 @@ def test_dot(self):
 
         with tm.assertRaisesRegexp(ValueError, 'aligned'):
             df.dot(df2)
+
+
+@pytest.fixture
+def df_duplicates():
+    return pd.DataFrame({'a': [1, 2, 3, 4, 4],
+                         'b': [1, 1, 1, 1, 1],
+                         'c': [0, 1, 2, 5, 4]},
+                        index=[0, 0, 1, 1, 1])
+
+
+@pytest.fixture
+def df_strings():
+    return pd.DataFrame({'a': np.random.permutation(10),
+                         'b': list(ascii_lowercase[:10]),
+                         'c': np.random.permutation(10).astype('float64')})
+
+
+class TestNLargestNSmallest(object):
+
+    # ----------------------------------------------------------------------
+    # Top / bottom
+    @pytest.mark.parametrize(
+        'n, order',
+        product(range(1, 11),
+                [['a'],
+                 ['c'],
+                 ['a', 'b'],
+                 ['a', 'c'],
+                 ['b', 'a'],
```

Comment:
@RogerThomas you will see this test failing for the below. The issues is that if one of the columns you are n* on is object this will fail. So I think that should raise a nice error message (you can check if *any* are object upfront).

```
___________________________________________________________________________________________ TestNLargestNSmallest.test_n[10-order119] ___________________________________________________________________________________________
pandas/tests/frame/test_analytics.py:1945: in test_n
    result = df.nsmallest(n, order)
pandas/core/frame.py:3478: in nsmallest
    return algorithms.select_n_frame(self, columns, n, 'nsmallest', keep)
pandas/core/algorithms.py:966: in select_n_frame
    values = getattr(series, method)(n, keep=keep)
pandas/core/series.py:1907: in nsmallest
    method='nsmallest')
pandas/core/algorithms.py:915: in select_n_series
    raise TypeError("Cannot use method %r with dtype %s" % (method, dtype))
E   TypeError: Cannot use method 'nsmallest' with dtype object
```